src/cargo/util/toml/mod.rs

```diff
@@ -1347,42 +1646,158 @@ impl TomlManifest {
             project.links.as_deref(),
         )?;
 
+        fn resolve_to_option<'a, T>(
+            features: &Features,
+            value: Option<MaybeWorkspace<T>>,
+            label: &str,
+            inheritable_fields: Option<&'a InheritableFields>,
+            resolve: impl FnOnce(&InheritableFields) -> Option<T>,
+        ) -> CargoResult<Option<T>> {
+            match value {
+                None => Ok(None),
+                Some(mw) => {
+                    match mw.resolve(&features, label, || get_ws(inheritable_fields), resolve) {
+                        Ok(t) => Ok(Some(t)),
+                        Err(e) => Err(e),
+                    }
+                }
+            }
+        }
+
         let metadata = ManifestMetadata {
-            description: project.description.clone(),
-            homepage: project.homepage.clone(),
-            documentation: project.documentation.clone(),
+            description: resolve_to_option(
+                &features,
+                project.description.clone(),
+                "description",
+                inheritable,
+                |i| i.description(),
+            )?,
+            homepage: resolve_to_option(
+                &features,
+                project.homepage.clone(),
+                "homepage",
+                inheritable,
+                |i| i.homepage(),
+            )?,
+            documentation: resolve_to_option(
+                &features,
+                project.documentation.clone(),
+                "documentation",
+                inheritable,
+                |i| i.documentation(),
+            )?,
             readme: readme_for_project(package_root, project),
-            authors: project.authors.clone().unwrap_or_default(),
-            license: project.license.clone(),
-            license_file: project.license_file.clone(),
-            repository: project.repository.clone(),
-            keywords: project.keywords.clone().unwrap_or_default(),
-            categories: project.categories.clone().unwrap_or_default(),
-            badges: me.badges.clone().unwrap_or_default(),
+            authors: resolve_to_option(
+                &features,
+                project.authors.clone(),
+                "authors",
+                inheritable,
+                |i| i.authors(),
+            )?
+            .unwrap_or_default(),
+            license: resolve_to_option(
+                &features,
+                project.license.clone(),
+                "license",
+                inheritable,
+                |i| i.license(),
+            )?,
+            license_file: resolve_to_option(
+                &features,
+                project.license_file.clone(),
+                "license_file",
+                inheritable,
+                |i| i.license_file(),
+            )?,
```

Comment:
Is `i.license_file)()` already an absolute path by this point?

Reply:
I don't think it's an absolute path by this point. I think it technically needs special handling to make this a path as we resolve it, but that is all should be done in part 2 or 3

Reply:
Could you note this explicitly in the PR description so its easier to bring others up to speed on whats included and whats not?

Reply:
will do!